pluto.md

```diff
@@ -0,0 +1,1280 @@
+This guide should get you started with everything you need for rewriting your Plutus Tx validators in Pluto, from scratch. No prerequisites are required – other than sufficient familiarity with Haskell and Plutus. Of course, you should know what Pluto *is*. Having a go through the syntax description also helps, but is not necessary.
+
+<details>
+<summary> Table of Contents </summary>
+
+- [Preamble: Why Pluto?](#preamble-why-pluto)
+- [Running & Testing Pluto](#running--testing-pluto)
+- [Syntax & Usage](#syntax--usage)
+  - [Lambda](#lambda)
+  - [Constants](#constants)
+    - [Boolean](#boolean)
+    - [Integer](#integer)
+    - [ByteString](#bytestring)
+    - [Text/String](#textstring)
+    - [Unit](#unit)
+    - [Data](#data)
+    - [Data Literals](#data-literals)
+      - [Integer constant - `42`](#integer-constant---42)
+      - [ByteString constant - `0x41`](#bytestring-constant---0x41)
+      - [List of data literals - `[1, 2, 3]`](#list-of-data-literals---1-2-3)
+      - [Map of data literal keys, to data literal values - `{ 1 = 0x42, 0xfe = 42 }`](#map-of-data-literal-keys-to-data-literal-values----1--0x42-0xfe--42-)
+      - [Sigma](#sigma)
+  - [Variables](#variables)
+  - [Plutus Core Builtin functions](#plutus-core-builtin-functions)
+  - [Infix Application](#infix-application)
+  - [If-then-else](#if-then-else)
+  - [Let bindings](#let-bindings)
+  - [Delay and Force](#delay-and-force)
+- [Recursion](#recursion)
+- [Analyzing basic examples](#analyzing-basic-examples)
+  - [Echo](#echo)
+  - [Hello](#hello)
+- [Intermission: Loading Pluto into Haskell](#intermission-loading-pluto-into-haskell)
+- [Using Pluto within Haskell](#using-pluto-within-haskell)
+- [Working with Builtin Lists](#working-with-builtin-lists)
+- [Working with Builtin Pairs](#working-with-builtin-pairs)
+- [Intermission: What is Data/BuiltinData?](#intermission-what-is-databuiltindata)
+- [Intermission: UPLC is strict](#intermission-uplc-is-strict)
+- [Writing a real Validator](#writing-a-real-validator)
+  - [Pre-requisites](#pre-requisites)
+  - [Plutus Tx](#plutus-tx)
+  - [Pluto](#pluto)
+  - [Testing and Benchmarks](#testing-and-benchmarks)
+- [Thumb rules for auditing/understanding/writing Pluto](#thumb-rules-for-auditingunderstandingwriting-pluto)
+- [Common Issues](#common-issues)
+  - [Using operators without explicit parens causes weird behavior](#using-operators-without-explicit-parens-causes-weird-behavior)
+  - [Unexpected `\` near a lambda](#unexpected--near-a-lambda)
+  - [Unexpected `in`](#unexpected-in)
+  - [Unexpected `=` within `let` binding](#unexpected--within-let-binding)
+  - [`UnexpectedBuiltinTermArgumentMachineError` - A builtin received a term argument when something else was expected](#unexpectedbuiltintermargumentmachineerror---a-builtin-received-a-term-argument-when-something-else-was-expected)
+- [Extra: Useful Links](#extra-useful-links)
+</details>
+
+# Preamble: Why Pluto?
```

Comment:
Just noticed this. Could you increase the heading level throughout this file? Look at other `.md` files for comparison. Each `.md` file is expected to use level-2+ headers (and an optional *single* level-1 header at the top).

Reply:
I originally did not want to do this since there's quite a bit heading nesting - like [these](https://github.com/TotallyNotChase/plutonomicon/blob/pluto/pluto.md#integer-constant---42).